src/Microsoft.AspNet.WebHooks.Receivers.Stripe/WebHooks/StripeWebHookReceiver.cs

```diff
@@ -148,75 +113,116 @@ public override string Name
             }
         }
 
-        /// <inheritdoc />
-        public void Dispose()
-        {
-            Dispose(true);
-            GC.SuppressFinalize(this);
-        }
-
         internal static bool IsTestEvent(string notificationId)
         {
             return string.Equals(TestId, notificationId, StringComparison.OrdinalIgnoreCase);
         }
 
         /// <summary>
-        /// Releases the unmanaged resources and optionally releases the managed resources.
+        /// Verifies that the signature header matches that of the actual body.
         /// </summary>
-        /// <param name="disposing"><c>true</c> to release both managed and unmanaged resources; <b>false</b> to release only unmanaged resources.</param>
-        protected virtual void Dispose(bool disposing)
+        protected virtual async Task<bool> VerifySignature(HttpRequestMessage request, string id)
         {
-            if (!_disposed)
+            // 1. Ensure configuration for this receiver and id exists.
+            var secretKey = await GetReceiverConfig(request, Name, id, SecretMinLength, SecretMaxLength);
+
+            // 2. Get the timestamp and expected signatures from the signature header. Header is a comma-separated set
+            // of key=value pairs.
+            var signatureHeaderValue = GetRequestHeader(request, SignatureHeaderName);
+            var pairs = signatureHeaderValue.SplitAndTrim(',');
+            string timestamp = null;
+            var expectedSignatures = new List<string>();
+            foreach (var pair in pairs)
             {
-                _disposed = true;
-                if (disposing)
+                var keyValuePair = pair.SplitAndTrim('=');
+                if (keyValuePair.Length != 2)
                 {
-                    if (_httpClient != null)
+                    var message = string.Format(
+                        CultureInfo.CurrentCulture,
+                        StripeReceiverResources.Receiver_InvalidHeaderFormat,
+                        SignatureHeaderName);
+                    request.GetConfiguration().DependencyResolver.GetLogger().Error(message);
+
+                    var invalidHeader = request.CreateErrorResponse(HttpStatusCode.BadRequest, message);
+                    throw new HttpResponseException(invalidHeader);
+                }
+
+                if (!string.IsNullOrEmpty(keyValuePair[1]))
```

Comment:
Is there a valid where this can happen? Should this be an assert?

Reply:
I was mistaken about `SplitAndTrim(...)`. But, I lean toward just removing this check.